packages/config-plugins/src/android/Scheme.ts

```diff
@@ -48,14 +51,15 @@ export function setScheme(
     return androidManifest;
   }
 
-  // Get the current schemes and remove them.
+  // Get the current schemes and remove them from the list of schemes to add.
   const currentSchemes = getSchemesFromManifest(androidManifest);
   for (const uri of currentSchemes) {
```

Comment:
So, I'm not an expert in the code base, but if I'm reading this right this is basically: "Don't add schemes that are already in the manifest". This means that existing schemes won't get updated (I don't even know if schemes can be updated, so this may be a non-issue).

What if we removed all schemes that the scheme plugin is generating and then regenerated them. I think it would have a similar effect, with the added bonus that we could update existing schemes (if that's even a thing that's possible).

```suggestion
  for (const uri of schemes) {
      androidManifest = removeScheme(uri, androidManifest);
```

Reply:
So instead of `Don't add existing schemes` we would just update schemes the `Scheme` plugin knows about. In addition, it would be interesting to see if we could handle this deduplication login in the `IntentFilters` plugin (which I'm pretty sure this depends on).

Reply:
In the **future** when a critical mass of packages in the ecosystem have their own plugin, we can support two different workflows:

1. bare workflow: you prebuild/eject once and make custom modifications to your `ios`, and `android` folders. Of course, most plugins aren't very destructive and can allow for users to re-run prebuild, but this is just a convenient side-effect and YMMV.
2. a new "custom" managed workflow: you `prebuild` your app as many times as you want and use it to add and customize native features using a unified TS/JS/JSON interface, in this workflow, the `ios`, and `android` folders are treated as disposable generated folders which can have files reset between prebuilds (this is when existing schemes would be removed) think of it kinda like a bundler for native projects. This workflow was designed mainly for more robust managed workflow cloud builds with [eas build](https://docs.expo.io/eas/) (to supersede `expo build`) so rerunning prebuild locally today isn't efficient enough to create a very pleasant developer experience. 

With this in mind, plugins should only add and modify content, they should not try to clean up modifications made in previous runs. This rule allows plugins with overlapping functionality to cooperate.

For example; if you have a couple authentication plugins which append custom schemes, they can create a combined list of schemes. But if just one of those plugins attempts to remove schemes that it assumes got added in a previous run, it could potentially break the other plugins and cause unexpected behavior.

Reply:
We're actually doing something somewhere in between the two options (possibly closer to 2) in a project at the moment. We have one project but based on some config files and environment variables, we can build either of two similar apps.

Unfortunately, we have to keep the generated Android folder around in order to configure our signing key, so we don't purely do option two, which would be much more comfortable.

Sorry for the off topic ramble - but it's pretty cool what you can do with expo :)

> For example; if you have a couple authentication plugins which append custom schemes, they can create a combined list of schemes. But if just one of those plugins attempts to remove schemes that it assumes got added in a previous run, it could potentially break the other plugins and cause unexpected behavior.

I completely agree with this. However, my concern (which may not be valid) is that the Scheme plugin won't update the manifest at all, if you change something in the config. 

Say, for example, the expected output of the plugin this run is
```xml
        <data android:scheme="com.example" some-other:thing="stuff" />
```

but last run it had generated
```xml
        <data android:scheme="com.example" />
```

The manifest wouldn't be updated at all. In this case (it would see that the scheme already exists on the Manifest, and skip generating it). Perhaps this is desired behavior (perhaps it was the user who removed the `some-other:thing` attribute), but I'd argue that as this plugin is responsible for generating the `com.example` scheme, it's safe for the plugin to update it.

This behavior would be in between the old behavior (nuke all schemes) and  the behavior in the PR (don't touch any schemes that already exist). Instead the behavior would be to nuke all schemes that we are going to generate (ensuring they are up to date). This wouldn't affect schemes generated by other plugins, unless they were using the same scheme as something that the scheme plugin is generating (in which case I suspect there will be other problems).

Sorry for the wall of test 😆

Reply:
Thanks for the insight, it's cool to hear that you're using a similar approach to what we've been working towards! The `withScheme` plugin's job is to take `scheme` from the Expo config and ensure the Android app conforms to them. I think keeping the plugin functionality simple (i.e. no extra attributes unless they're defined in the ExpoConfig schema). 

To customize the schemes more, you could create a new mod plugin that adds more custom functionality by passing it custom properties in the app.config.js:

`app.config.js`
```ts
export default {
  plugins: [
    [
      withCustomSchemes, [
      { scheme: 'com.example', 'some-other:thing': 'stuff' },
      ] 
    ]
  ]
}
```

`withCustomSchemes.js`
```js
const {withAndroidManifest} = require('@expo/config-plugins');
module.exports = (config, customSchemes) => {
  return withAndroidManifest(config, config => {
    // Modify config.modResults as an AndroidManifest
    return config;
  }) 
}
```